commands/login.go

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) OpenFaaS Project 2017. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package commands
+
+import (
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/openfaas/faas-cli/config"
+	"github.com/spf13/cobra"
+)
+
+var (
+	username      string
+	password      string
+	passwordStdin bool
+)
+
+func init() {
+	loginCmd.Flags().StringVar(&gateway, "gateway", defaultGateway, "Gateway URI")
+	loginCmd.Flags().StringVarP(&username, "username", "u", "", "Gateway username")
+	loginCmd.Flags().StringVarP(&password, "password", "p", "", "Gateway password")
+	loginCmd.Flags().BoolVar(&passwordStdin, "password-stdin", false, "Reads the gateway password from stdin")
+
+	faasCmd.AddCommand(loginCmd)
+}
+
+var loginCmd = &cobra.Command{
+	Use:   `login [--username USERNAME] [--password PASSWORD] [--gateway GATEWAY_URL]`,
+	Short: "Log in to OpenFaaS gateway",
+	Long:  "Log in to OpenFaaS gateway.\nIf no gateway is specified, the default local one will be used.",
+	Example: `  faas-cli login -u user -p password --gateway http://localhost:8080
+  cat ~/faas_pass.txt | faas-cli login -u user --password-stdin --gateway https://openfaas.mydomain.com`,
+	RunE: runLogin,
+}
+
+func runLogin(cmd *cobra.Command, args []string) error {
+
+	if len(username) == 0 {
+		return errors.New("must provide --username or -u")
+	}
+
+	if len(password) > 0 {
+		fmt.Println("WARNING! Using --password is insecure, consider using: cat ~/faas_pass.txt | faas-cli login -u user --password-stdin")
+		if passwordStdin {
+			return errors.New("--password and --password-stdin are mutually exclusive")
+		}
+
+		if len(username) == 0 {
+			return errors.New("must provide --username with --password")
+		}
+	}
+
+	if passwordStdin {
+		if len(username) == 0 {
+			return errors.New("must provide --username with --password-stdin")
+		}
+
+		passwordStdin, err := ioutil.ReadAll(os.Stdin)
+		if err != nil {
+			return err
+		}
+
+		password = strings.TrimSpace(string(passwordStdin))
```

Comment:
Do we also need to trim the --password flag data?

Reply:
done in the latest commit